lib/auth/default_current_user_provider.rb

```diff
@@ -56,6 +79,7 @@ class Auth::DefaultCurrentUserProvider
   def initialize(env)
     @env = env
     @request = Rack::Request.new(env)
+    @cookie_jar = ActionDispatch::Request.new(env).cookie_jar
```

Comment:
I don't think we need `Rack::Request.new(env)` anymore since `ActionDispatch::Request` has some sort of methods parity with `Rack::Request`.

Reply:
FWIW, if we only need the cookie jar, it is already present in the `env`. `env["action_dispatch.cookies"]`